draft-ietf-quic-transport.md

```diff
@@ -1928,8 +1928,10 @@ consider the peer address to have been validated.
 
 Additionally, an endpoint MAY consider the peer address validated if the peer
 uses a connection ID chosen by the endpoint and the connection ID contains at
-least 64 bits of entropy.  This means that any valid packet received by the
-client validates the server address.
+least 64 bits of entropy. A client can consider the server address validated on
+successfully processing any packet received from the server, since for
```

Comment:
Rather than "any packet", I think we might wan to say "Initial packet"? By stating as such, we can be clear that the receipt of Retry packets does not mean that the path has been validated. We do not need to be worried about Handshake or 1-RTT packets because processing an Initial packet is the prerequisite for obtaining keys necessary to decrypt Handshake and 1-RTT packets.

Reply:
So any packet is correct, if we also say "successfully processed".  Retry and VN both echo the Destination Connection ID field, and Initial packets use encryption.  This is why I think that I will roll back Jana's proposed change - at least partly.

Reply:
@martinthomson Re Retry, I'm not sure if that is the case.

[Section 17.2.5.1](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#section-17.2.5.1-2) specifies that a server MUST NOT echo the value of the DCID field of an Initial packet that it receives. Therefore, for a client, receiving a Retry packet does not mean that the server has received the Initial packet carrying the DCID that the client chose.

Reply:
Oh right.  It's the authentication tag in Retry that provides that proof.  Such a complicated, fiddly protocol.

Reply:
Ah right. Receipt of Retry packets do validate the address using a different mechanism 🤦